servers/services/src/main/java/org/projectnessie/services/rest/TreeResourceWithAuthorizationChecks.java

```diff
@@ -135,8 +133,12 @@ protected void checkHashOnRef(String refName, String hash) throws NessieNotFound
         // Special case, mostly for tests:
         // When the default branch is deleted, we cannot verify whether the branch already exists
         // (it does not, so the check would fail).
-        if (refName.equals(getConfig().getDefaultBranch()) && hash == null
-            || hash.equals(getStore().noAncestorHash().asString())) {
+        System.out.println("EX refName = " + refName);
+        System.out.println("EX hash = " + hash);
+        System.out.println("getStore().noAncestorHash() = " + getStore().noAncestorHash());
+
+        if (Objects.equals(getConfig().getDefaultBranch(), refName) && hash == null
+            || Objects.equals(getStore().noAncestorHash().asString(), hash)) {
           // prevent the "hash-on-ref" check, but still do the "can-view" check
           hash = null;
           return;
```

Comment:
The `return` needs to be removed (sorry, oversight)

Reply:
I'll have a follow-up PR where I'll merge this `hashOnRef` check with `namedHashOrThrow` as both are essentially doing the same